### DISCUSSION TITLE
close position by stop or take profit

### DISCUSSION THREAD

**Discussion by Brianramirez27** [kernc/backtesting.py#462] [General]
Hello, I programmed a strategy of crossings of emas and established a sl and tp but it only opens 3 operations and does not close them, I would like to know if I should do something to close the operations by tp or sl
![pregunta final](https://user-images.githubusercontent.com/84251018/132721117-1ed60097-fb2c-4612-8c99-f2f398697c08.JPG)

**Comment by kernc** [comment]
SL and TP orders should be executed automatically once the price level (0.2 or 4.0, for example) is hit.

**Comment by Brianramirez27** [comment]
I only open a trade and never close the operation in 10 years for what reason can it be
![Uploading 45.JPG…]()

**Reply by kernc** [reply]
I don't know your data, and you wholly fail to provide a [reproducible code sample](https://github.com/kernc/backtesting.py/blob/master/CONTRIBUTING.md#issues). But you set the `sl=` and `tp=` price points to `0.2` and `4.0`, and you should note that those are absolute currency (e.g. USD) rather than relative values.

**Reply by Brianramirez27** [reply]
The estop is established in points, in percentage, or at the price of the instrument that I want to close

**Reply by kernc** [reply]
[At the price](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Order.sl) only. Is there anything in the documentation suggesting otherwise?

**Reply by Brianramirez27** [reply]
Not in the documentation it does not say that, only that I do not see much information on how to place the stop and the tp, since if it goes in price how could it be to know at what price there is an entry and what price should the stop go, I supposed It was in percentage since in each entry that was only if it reached that percentage it closed the position, but I see that it is not like that, then I must know at what price to place the stop and if there are a thousand operations I must know each operation at what price is the stop?

**Reply by Brianramirez27** [reply]
![ffff](https://user-images.githubusercontent.com/84251018/133797150-f3224f93-5439-4c6c-b3c0-265e6a915f98.JPG)
These are the prices that I have, which would be the correct way to put a stop and a take profi

**Reply by kernc** [reply]
Were you aware of and have you seen [the tutorials](https://kernc.github.io/backtesting.py/doc/backtesting/index.html#tutorials)? [At least](https://kernc.github.io/backtesting.py/doc/examples/Multiple%20Time%20Frames.html) [two of those](https://kernc.github.io/backtesting.py/doc/examples/Trading%20with%20Machine%20Learning.html) show how to set `sl=` (...) properly ... :smile:

**Reply by Brianramirez27** [reply]
I already solved it, the current price has to go first and multiply it by the tp and the stop

if crossover(self.ma1, self.ma2):
            self.buy(sl= price*0.99,tp=price* 1.01)

**Comment by Brianramirez27** [comment]
?????
